[DOCS] Kafka Connect and Mirror Maker upgrades, plus minor style edits (#48)
rh-messaging/amq-streams-docs: books/assembly-upgrade-1-1-0.adoc

|1.1.0                 |2.1.1
|=======================

> Although {ProductName} 1.1.0 and Kafka 2.1.1 are both minor releases, the Kafka protocol has changed since Kafka 2.0.0 was released. In particular, the message format version and inter-broker protocol version are both now at version 2.1. As a result, the upgrade process involves making configuration changes to existing Kafka brokers and code changes to client applications (producers, consumers, Kafka Connect, and MirrorMaker). The following table shows the differences between the two Kafka versions:

Reviewer: Seems a bit inconsistent ... why _Kafka Connect_ but not _Kafka Mirror Maker_?

> Although Kafka 2.0.0 and 2.1.1 use the same version of Zookeeper, Red Hat recommends that you update your Zookeeper cluster to use the newest Zookeeper binaries before proceeding with the main {ProductName} 1.1.0 upgrade.

Reviewer 2: ```suggestion
Although Kafka 2.0.0 and 2.1.1 use the same version of Zookeeper, it is recommended that you update your Zookeeper cluster to use the newest Zookeeper binaries before proceeding with the main {ProductName} 1.1.0 upgrade.
```

.. xref:proc-updating-zookeeper-binaries-{context}[Updating the Zookeeper binaries]

. Second, upgrade all Kafka brokers to {ProductName} 1.1.0 and configure them to use the previous protocol versions.

.. xref:proc-upgrading-kafka-brokers-to-amq-streams-1-1-0-{context}[Upgrading Kafka brokers to {ProductName} 1.1.0]

> . Third, upgrade all Kafka brokers and client applications (producers, consumers, Kafka Connect, and MirrorMaker) to Kafka 2.1.1. To avoid cluster downtime, this stage involves:

Reviewer: Same as above.

Reviewer 3: My only question would be whether we need to repeat the same information in brackets in three places. Maybe we could say it once as something like `Client applications might include...` in Strategies for Upgrading Clients. If we need separate upgrade procedures for Connect and Mirror Maker, as Jakub suggests, then we'll need to reorganise this assembly. (Note, I'm working on the same content for PR #1528)

Author: I've gone with Paul's suggestion. Changed to Kafka MirrorMaker.
